google/cloud/bigtable/backup.py

```diff
@@ -0,0 +1,332 @@
+
+
+class Backup(object):
+    """Representation of a Google Cloud Bigtable Backup.
+
+    A :class: `Backup` can be used to:
+
+    * :meth:`create` the backup
+    * :meth:`update` the backup
+    * :meth:`delete` the backup
+
+    :type backup_id: str
+    :param backup_id: The ID of the backup.
+
+    :type instance: :class:`~google.cloud.spanner_v1.instance.Instance`
+    :param instance: The Instance that owns this Backup.
+
+    :type cluster_id: str
+    :param cluster_id: (Optional) The ID of the Cluster that contains this Backup.
+                       Required for calling 'delete', 'exists' etc. methods.
+
+    :type table_id: str
+    :param table_id: (Optional) The ID of the Table that the Backup is for.
+                     Required if the 'create' method will be called.
+
+    :type expire_time: :class:`datetime.datetime`
+    :param expire_time: (Optional) The expiration time after which the Backup
+                        will be automatically deleted. Required if the `create`
+                        method will be called.
+    """
+
+    def __init__(
+        self, backup_id, instance, cluster_id=None, table_id=None, expire_time=None
+    ):
+        self.backup_id = backup_id
+        self._instance = instance
+        self._cluster = cluster_id
+        self.table_id = table_id
+        self._expire_time = expire_time
+
+        self._parent = None
+        self._source_table = None
+        self._start_time = None
+        self._end_time = None
+        self._size_bytes = None
+        self._state = None
+
+    @property
+    def name(self):
+        """Backup name used in requests.
+
+        The Backup name is of the form
+
+            ``"projects/../instances/../clusters/../backups/{backup_id}"``
+
+        :rtype: str
+        :returns: The Backup name.
+
+        :raises: ValueError: If the 'cluster' has not been set.
+        """
+        if not self._cluster:
+            raise ValueError('"cluster" parameter must be set')
+
+        return "{}/clusters/{}/backups/{}".format(
+            self._instance.name, self._cluster, self.backup_id
+        )
+
+    @property
+    def cluster(self):
+        """The ID of the [parent] cluster used in requests.
+
+        :rtype: str
+        :returns: The ID of the cluster containing the Backup.
+        """
+        return self._cluster
+
+    @cluster.setter
+    def cluster(self, cluster_id):
+        self._cluster = cluster_id
+
+    @property
+    def parent(self):
+        """Name of the parent cluster used in requests.
+
+        .. note::
+          This property will return None if ``cluster`` is not set.
+
+        The parent name is of the form
+
+            ``"projects/{project}/instances/{instance_id}/clusters/{cluster}"``
+
+        :rtype: str
+        :returns: A full path to the parent cluster.
+        """
+        if not self._parent and self._cluster:
+            self._parent = self._instance.name + "/clusters/" + self._cluster
```

Comment:
nit.
Can use [cluster_path](https://github.com/googleapis/python-bigtable/blob/5c1d61827618d254c453b3871c0022a8d35bfbb2/google/cloud/bigtable_admin_v2/gapic/bigtable_table_admin_client.py#L110) as well here. It will verify presence of all components as well.

Reply:
:+1:

Reply:
Regarding verification, the same argument as above.

Reply:
BTW, found and fixed the same occurrence in `table.list_backups`, that is added `cluster_path` there as well.